### DISCUSSION TITLE
How to implement  dynamic manage service  during the main service running?

### DISCUSSION THREAD

**Discussion by takkuumi** [tokio-rs/axum#2195] [Q&A]
background:
I need to manage a lot of hardware connections (via MODBUS), because in some cases, I will dynamically remove or add devices. I want to implement each connection with a service (because their functions are consistent), and then notify the main service through the channel to create and load or delete and unload a connection service. I don't know if I can use AXUM to achieve it, the service is implemented using rpc (tonic).

Lessons learned:
1. Why not implement it with a connector manager?
Already tried it, with normal network, it works fine. The real situation is that the target network jitters frequently, and continuous attempts to connect or timeout retries will slow down the efficiency of the entire application, because the connector is locked and wrapped, and the efficiency cannot meet the customer's requirements for timeliness.

**Comment by davidpdrsn** [comment]
axum doesn’t deal the connection handling. You’ll probably wanna use hyper’s low level connection API.
